src/lib/io/sol-gpio-common.c

```diff
@@ -74,3 +77,39 @@ sol_gpio_open(uint32_t pin, const struct sol_gpio_config *config)
 
     return gpio;
 }
+
+SOL_API enum sol_gpio_edge
+sol_gpio_edge_from_str(const char *edge)
+{
```

Comment:
I don't have a problem with doing the conversion from string to enum and vice versa here, but wouldn't it be easier to do this in the high-level JS API? After all, the soletta C API has no concept of strings that refer to edges. It only has the enum. Thus, the conversion from string to enum and vice versa should happen in the high-level JS API.